lib/system/connection_probe.rb

```diff
@@ -8,12 +10,12 @@ class ConnectionProbe < ActiveRecord::Base
 
       class << self
         def connection_config
-          spec = configuration_specification.config.dup
-          if oracle?
-            spec[:password] = ENV.fetch('ORACLE_SYSTEM_PASSWORD') {|key| raise KeyError, "Environment #{key} is mandatory"}
-            spec[:username] = 'SYSTEM'
+          configuration_specification.config.dup.tap do |spec|
+            if oracle? && ENV.fetch('ORACLE_SYSTEM_PASSWORD', nil).present?
```

Comment:
There are two ENV.fetch done for the env var, maybe extract it to a variable?

Also, maybe it could just be `ENV['ORACLE_SYSTEM_PASSWORD']`, do we really need to set a default?